src/utils.cpp

```diff
@@ -165,6 +165,42 @@ const char *utils::guesspath(char *buf, int len, FILE *fp)
   return buf;
 }
 
+// read line into buffer. if line is too long keep reading until EOL or EOF
+// but return only the first part with a newline at the end.
+
+char *utils::fgets_trunc_nl(char *buf, int size, FILE *fp)
```

Comment:
I would just call this `fgets_trunc` without the `nl`. First of all it reads like `n1` and it doesn't add new information compared to the default behavior of `fgets`.

Reply:
> I would just call this `fgets_trunc` without the `nl`. First of all it reads like `n1` and it doesn't add new information compared to the default behavior of `fgets`.

I agree on the n1/nl thing, but it **does** indicate an additional difference in behavior. `fgets()` will add the terminating NULL, but this function **also** guarantees a newline character at the end. (which is what was initially injected by `utils::read_lines_from_file()` into the block of lines that is then broadcast so it can be split and parsed concurrently on all MPI ranks)

Reply:
made the rename